lib/Doctrine/ORM/Query/Expr.php

```diff
@@ -41,8 +41,8 @@ class Expr
      *     // (u.type = ?1) AND (u.role = ?2)
      *     $expr->andX($expr->eq('u.type', ':1'), $expr->eq('u.role', ':2'));
      *
-     * @param Expr\Comparison|Expr\Func|Expr\Orx|string $x Optional clause. Defaults to null, but requires at least one
-     *                                                     defined when converting to string.
+     * @param mixed $x Optional clause. Defaults to null, but requires
```

Comment:
```suggestion
     * @param Expr\Comparison|Expr\Func|Expr\Orx|Expr\Andx|string $x Optional clause. Defaults to null, but requires
```

Reply:
hmm. other functuions in this class has `mixed` param annotaion. Maybe there must be equal `mixed`?

Reply:
If there is a finite number of types that can be accepted here, I'd rather document them. This will help SA tools spot bugs.

Reply:
Ok, im do this later